sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -350,6 +350,9 @@ object QueryTest {
     Row.fromSeq(row.toSeq.map {
       case null => null
       case d: java.math.BigDecimal => BigDecimal(d)
+      // Round up the values to avoid approximate error
+      case f : Float => BigDecimal(f).setScale(8, BigDecimal.RoundingMode.HALF_UP).toFloat
```

Comment:
@davies, I am still not sure in which case, 1111111111111111.1 needs to be rounded up? In my opinion, this value does not need to do so. In my test cases, compare 0.1 + 0.2 with 0.3 will fail the test cases, but it only generate a very small error, it won't be difference of 0.1. Can you give me a specific case so I can test and fix my codes?